crates/tracing/src/lib.rs

```diff
@@ -0,0 +1,220 @@
+//! Miralis ABI
+//!
+//! While miralis forwards standard SBI calls to the virtualized firmware, miralis do expose its own
+//! ABI for the firmware to interact with.
+//!
+//! The Miralis ABI tries to be compatible with the SBI specification as best as it can.
+//! See: https://github.com/riscv-non-isa/riscv-sbi-doc
+#![no_std]
+
+use core::arch::{asm, global_asm};
+
+use miralis_abi::failure;
+
+pub fn print_statistics(stats: Statistics) {
+    log::info!("{:?}", stats);
+}
+
+#[macro_export]
+macro_rules! measure {
+    ($policy_name:literal, $out:literal, $in:expr, $in2:expr) => {
+        let mut values: [usize; NB_REPEATS] = [0; NB_REPEATS];
+
+        for i in 0..NB_REPEATS {
+            values[i] = $in();
+        }
+
+        let stats = get_statistics(values);
```

Comment:
This macro uses locally-defined function, so it will fail with a compile error when used by another crate or module. There is a simple way to fix this, which is to use an absolute path for each local function used. There is a magic variable `$crate` to do that:

```rs
let stats = $crate::get_statistics(values)
```

With the above version the macro can be used without importing the `get_statistics` function. In general macros in libraries should encapsulate all of their import so that they work like standard functions.